src/Compilers/Core/CodeAnalysisTest/Collections/SegmentedArrayHelperTests.cs

```diff
@@ -12,6 +17,90 @@ namespace Microsoft.CodeAnalysis.UnitTests.Collections
 {
     public class SegmentedArrayHelperTests
     {
+        [StructLayout(LayoutKind.Sequential, Size = 2)]
+        private struct Size2 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 4)]
+        private struct Size4 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 8)]
+        private struct Size8 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 12)]
+        private struct Size12 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        private struct Size16 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 24)]
+        private struct Size24 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 28)]
+        private struct Size28 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 32)]
+        private struct Size32 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 40)]
+        private struct Size40 { }
+
+        public static IEnumerable<object[]> ExplicitSizeTypes
+        {
+            get
+            {
+                yield return new object[] { typeof(Size2) };
+                yield return new object[] { typeof(Size4) };
+                yield return new object[] { typeof(Size8) };
+                yield return new object[] { typeof(Size12) };
+                yield return new object[] { typeof(Size16) };
+                yield return new object[] { typeof(Size24) };
+                yield return new object[] { typeof(Size28) };
+                yield return new object[] { typeof(Size32) };
+                yield return new object[] { typeof(Size40) };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ExplicitSizeTypes))]
+        public void ExplicitSizesAreCorrect(Type type)
+        {
+            var unsafeSizeOfMethod = typeof(Unsafe).GetMethod(nameof(Unsafe.SizeOf)).MakeGenericMethod(type);
+            var size = (int)unsafeSizeOfMethod.Invoke(null, null);
```

Comment:
>(int)unsafeSizeOfMethod.Invoke(null, null); [](http://example.com/codeflow?start=23&length=43)

Use ```InvokeUnsafeSizeOf``` here?

Reply:
➡️ Yep, amended the last commit to include this call